### ISSUE TITLE
Build Ceph-CSI images in CentOS CI OpenShift for job speedup

### ISSUE THREAD

**Issue by nixpanic** [ceph/ceph-csi#1628]
# Describe the feature you'd like to have #

The CentOS CI is running inside OpenShift and can be used to build (and cache) container images. The ceph-csi:test image is heavily used during CI jobs, and does not change often (the `scripts/Dockerfile.test` file). Jobs that build and run the ceph-csi:test container image, can pull the existing image from the OpenShift envionment (if the PR does not change the `scripts/Dockerfile.test` file. This should reduce network bandwidth, and make the tests faster.

# What is the value to the developer? (why is it a priority?) #

Faster testing with certain CI jobs, quicker feedback on PRs.

# How will we know we have a good solution? (acceptance criteria) #

1. automated builds/updates of the ceph-csi:test image when PRs get merged
1. `make containerized-test` can pull the image from the local OpenShift repository

# Additional context #

A [BuildConfiguration](https://docs.openshift.com/container-platform/4.5/builds/understanding-buildconfigs.html) and ImageStream will be needed. As well as a CI job that rebuilds images after PR merges.

The OpenShift deployment where our Jenkins is hosted does not provide access to the registry for the bare-metal systems that run the CI jobs. This has been [reported to the CentOS team](https://pagure.io/centos-infra/issue/141) and is currently being investigated.

**Comment by nixpanic** [comment]
Minor hickup in the idea, the firewall in the CI environment does not allow bare-metal systems access to the default OpenShift registry: https://lists.centos.org/pipermail/ci-users/2020-October/002141.html

**Comment by nixpanic** [comment]
Report and research by the admins of the OpenShift environment can be found at https://pagure.io/centos-infra/issue/141

**Comment by Madhu-1** [comment]
@nixpanic are we planning to go with a workaround by deploying our own registry?

**Comment by nixpanic** [comment]
> @nixpanic are we planning to go with a workaround by deploying our own registry?

Yes, will try that to see if there is any improvement in the speed of the CI jobs

**Comment by nixpanic** [comment]
The `make containerized-build` commands in `mini-e2e.groovy`, `mini-e2e-helm.groovy` and `upgrade-tests.groovy` still build the `:devel` container for each run. These jobs need to be adapted to pull the image ad is done in #1656.
